rcp/org.testeditor.rcp4.views.projectexplorer/src/org/testeditor/rcp4/views/projectexplorer/ResourceDecorator.xtend

```diff
@@ -0,0 +1,68 @@
+package org.testeditor.rcp4.views.projectexplorer
+
+import org.eclipse.core.resources.IMarker
+import org.eclipse.core.resources.IProject
+import org.eclipse.core.resources.IResource
+import org.eclipse.core.resources.ResourcesPlugin
+import org.eclipse.core.runtime.CoreException
+import org.eclipse.jdt.core.IClasspathEntry
+import org.eclipse.jface.viewers.IDecoration
+import org.eclipse.jface.viewers.ILightweightLabelDecorator
+import org.eclipse.jface.viewers.LabelProvider
+import org.eclipse.jface.viewers.LabelProviderChangedEvent
+import org.eclipse.swt.widgets.Display
+import org.eclipse.ui.ISharedImages
+import org.eclipse.ui.internal.WorkbenchImages
+
+class ResourceDecorator extends LabelProvider implements ILightweightLabelDecorator {
+
+	static val NO_SEVERITY = -1
+
+	// only descriptors, need not be disposed => no cleanup necessary
+	val errorIcon = WorkbenchImages.getImageDescriptor(ISharedImages.IMG_DEC_FIELD_ERROR)
```

Comment:
I think using the WorkbenchImages static method makes it harder to test this class. We should use a WorkbenchImagesHelper class to avoid static calls.